File: 06_09-22/Maximum_Difference_Between_Increasing_Elements.py
from math import inf
class Solution:
    def maximumDifference(self, n: List[int]) -> int:
        zz=inf
        res=-1
        for i in n:
            zz=min(zz,i)
            if(i>zz):
                res=max(res,i-zz)
        return res
            

File: 06_09-22/Maximum_Difference_Between_Increasing_Elements.cpp
class Solution {
public:
    int maximumDifference(vector<int>& n) {
        int  zz=INT_MAX;
        int res=-1;
        for (int i=0;i<n.size();i++)
        {
            zz=min(zz,n[i]);
            if(n[i]>zz)
            {
                res=max(res,n[i]-zz);
            }
        }
        return res;
            
    }
};